### PR TITLE
[DatePicker] Labels

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### WHY are these changes introduced?

Internal issue linked below

#### Default - start of range
Screen Shot 2020-12-01 at 11 47 46 AM

#### Default - end of range
Screen Shot 2020-12-01 at 11 48 21 AM

#### Custom - start of range
Screen Shot 2020-12-01 at 12 03 16 PM

#### Custom - end of range
Screen Shot 2020-12-01 at 12 03 38 PM

### WHAT is this pull request doing?

* Add default start / end labels
* Add the ability of custom labels

## 

### How to 🎩

* Code sample of a custom dual date picker setup using prefix labels for ranges

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React, {useState, useCallback} from 'react';

import {Page, DatePicker, DisplayText} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <DatePickerExample />
    </Page>
  );
}

function DatePickerExample() {
  const [{month, year}, setDate] = useState({month: 1, year: 2018});
  const [selectedDates, setSelectedDates] = useState(
    new Date('Wed Feb 07 2018 00:00:00 GMT-0500 (EST)'),
  );
  const [selectedDatesTwo, setSelectedDatesTwo] = useState(
    new Date('Sat Feb 10 2018 00:00:00 GMT-0500 (EST)'),
  );

  const handleMonthChange = useCallback(
    (month, year) => setDate({month, year}),
    [],
  );

  const handleDateChange = (callback: (date: Date) => void) => ({
    start,
  }: {
    start: Date;
    end: Date;
  }) => callback(start);

  return (
    <>
      <DisplayText>Start</DisplayText>
      <DatePicker
        month={month}
        year={year}
        onChange={handleDateChange(setSelectedDates)}
        onMonthChange={handleMonthChange}
        selected={selectedDates}
        disableDatesBefore={new Date('Sat Feb 03 2018 00:00:00 GMT-0500 (EST)')}
        disableDatesAfter={new Date('Sun Feb 18 2018 00:00:00 GMT-0500 (EST)')}
        dayAccessibilityLabelPrefix="Custom start of range"
      />
      <DisplayText>End</DisplayText>
      <DatePicker
        month={month}
        year={year}
        onChange={handleDateChange(setSelectedDatesTwo)}
        onMonthChange={handleMonthChange}
        selected={selectedDatesTwo}
        disableDatesBefore={new Date('Sat Feb 03 2018 00:00:00 GMT-0500 (EST)')}
        disableDatesAfter={new Date('Sun Feb 18 2018 00:00:00 GMT-0500 (EST)')}
        dayAccessibilityLabelPrefix="Custom start (but also end) of range"
      />
    </>
  );
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: locales/en.json [hunk 1]
```diff
@@ -53,6 +53,8 @@
       "previousMonth": "Show previous month, {previousMonthName} {showPreviousYear}",
       "nextMonth": "Show next month, {nextMonth} {nextYear}",
       "today": "Today ",
+      "start": "Start of range ",
+      "end": "End of range",
```
Comment (alex-page) [inline locales/en.json:57]: Should we make these consistent and remove the space from `start`

File: src/components/DatePicker/DatePicker.tsx [hunk 2]
```diff
@@ -43,6 +43,8 @@ export interface DatePickerProps {
    * @default 0
    */
   weekStartsOn?: number;
+  /** Visually hidden prefix text for selected days on single selection date pickers */
+  dayAccessibilityLabelPrefix?: string;
```
Comment (alex-page) [inline src/components/DatePicker/DatePicker.tsx:47]: Should this be optional? If it makes the experience better for screen readers it should be required.

**Review by alex-page — COMMENTED** [review]
This looks great, thanks @AndrewMusgrave. I added some comments and still need to test it with voiceover.

**Review by kyledurand — APPROVED** [review]
LGTM, Alex made some valid points though

File: src/components/DatePicker/DatePicker.tsx (hunk 2 above)
Comment (AndrewMusgrave) [reply src/components/DatePicker/DatePicker.tsx:47]: It won't always be needed. When you have a single date picker, you won't need it. When you have a ranged date picker we have defaults (added in this pull request). But when you're creating a custom ranged date picker (see order list date filters) we need custom prefixes.
Screen Shot 2021-01-05 at 10 18 25 AM

**Review by alex-page — APPROVED** [review]

**Pull request merged**